src/serial.rs

```diff
@@ -0,0 +1,474 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Portions Copyright 2020 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Provides emulation for Linux serial console.
+//!
+//! This is done by emulating an UART 8250 serial port.
+
+use std::collections::VecDeque;
+use std::io::{Result, Write};
+
+use vmm_sys_util::eventfd::EventFd;
+
+// Register offsets.
+const DATA: u8 = 0;
```

Comment:
Can you add a comment here about the fact that data represents either RBR or THR depending on the performed operations (read/write)? This was briefly mentioned in the README.md as well, but I believe it's nice to have it somewhere in the code as well.